api/api/routers/tools_router.py

```diff
@@ -0,0 +1,56 @@
+from fastapi import APIRouter, HTTPException
+
+from api.dependencies.services import ToolsServiceDep
+from api.dependencies.task_info import TaskInfoDep
+from api.routers.view_models.tools_view_models import ToolParamsRequest, ToolResponse
+from core.storage import ObjectNotFoundException
+
+router = APIRouter(prefix="/agents/{task_id}/tools")
```

Comment:
Are we sure we want to link to an agent ? 
It might be better to have a `/tools` endpoint where tools can be attached to multiple agents ?

Reply:
See `# The endpoint is this router have a 'task' dependency that just checks that the task in url exists but the task is not used since we fetch all the tools from the tenant.`

The thing is, the specs originally attached the tools to a single agent, but I've proposed that tools to be shared among the whole tenant because it's good to reuse tools among agents which Pierre agreed upon. 

But the endpoint is on /tasks/ in case we ever want to revert this decision without updating the frontend. Does this make sense ?

Reply:
I think you can just remove the dependency to be honest

Reply:
done https://github.com/WorkflowAI/WorkflowAI/pull/204/commits/bd3a8684bb0d6be4d7e1d2d55fb3508993c44c5d